src/ast_editor.ml

```diff
@@ -321,9 +327,19 @@ let open_pp_doc instance ~document =
   match fetch_pp_code ~document with
   | Error e -> Promise.return (Error e)
   | Ok pp_pp_str ->
+    let file_name =
+      match Pp_path.get_kind ~document with
+      | Structure `Reason ->
+        String.chop_suffix_exn ~suffix:".re" (TextDocument.fileName document)
+        ^ ".ml"
+      | Signature `Reason ->
+        String.chop_suffix_exn ~suffix:".rei" (TextDocument.fileName document)
+        ^ ".mli"
+      | _ -> TextDocument.fileName document
```

Comment:
Since this relies on dune and isn't straight forward to understand reading the code, would you add a comment explaining that .ml is present at the same location as the .re file, etc?

Reply:
This part is related to creating the PP doc preview. It establishes a preview file with ocaml syntax, but as the current file can be a `.re`, I change it here to fit the ocaml syntax.
![Captura de Tela 2024-12-12 às 13 02 31](https://github.com/user-attachments/assets/6e9e3839-ae0f-4e22-9daa-572b8a4169fb)

As I said in [this comment](https://github.com/ocamllabs/vscode-ocaml-platform/pull/1685#discussion_r1882423107), later we can convert this content from ocaml syntax to reasonml, and this change of extension will not be needed anymore.